shap/plots/_style.py

```diff
@@ -0,0 +1,109 @@
+"""Configuration of customisable style options for SHAP plots.
+
+NOTE: This is experimental and subject to change!
+"""
+
+from __future__ import annotations
+
+from contextlib import contextmanager
+from dataclasses import dataclass, field, replace
+from typing import Union
+
+import numpy as np
+
+from . import colors
+
+# Type hints, adapted from matplotlib.typing
+RGBColorType = Union[tuple[float, float, float], str]
+RGBAColorType = Union[
+    str,  # "none" or "#RRGGBBAA"/"#RGBA" hex strings
+    tuple[float, float, float, float],
+    # 2 tuple (color, alpha) representations, not infinitely recursive
+    # RGBColorType includes the (str, float) tuple, even for RGBA strings
+    tuple[RGBColorType, float],
+    # (4-tuple, float) is odd, but accepted as the outer float overriding A of 4-tuple
+    tuple[tuple[float, float, float, float], float],
+]
+ColorType = Union[RGBColorType, RGBAColorType, np.ndarray]
+
+
+@dataclass
+class StyleConfig:
+    """Configuration of colors across all matplotlib-based shap plots."""
+
+    # Waterfall plot config
+    primary_color_positive: ColorType = field(default_factory=lambda: colors.red_rgb)
+    primary_color_negative: ColorType = field(default_factory=lambda: colors.blue_rgb)
+    secondary_color_positive: ColorType = field(default_factory=lambda: colors.light_red_rgb)
+    secondary_color_negative: ColorType = field(default_factory=lambda: colors.light_blue_rgb)
+    hlines_color: ColorType = "#cccccc"
+    vlines_color: ColorType = "#bbbbbb"
+    text_color: ColorType = "white"
+    tick_labels_color: ColorType = "#999999"
+
+
+def load_default_style() -> StyleConfig:
+    """Load the default style configuration."""
+    # In future, this could allow reading from a persistent config file, like matplotlib rcParams
+    return StyleConfig()
+
+
+# Singleton instance that determines the current style.
+# CAREFUL! Only access this instance from the `_style` namespace, do not import it directly.
+# This is to ensure that the correct instance is picked up when the STYLE object is
+# reassigned in the _style module.
+#
+#     from shap.plots import _style
+#     color = _style.STYLE.text_color       # Correct
+#
+#     from shap.plots._style import STYLE   # Wrong!
+#     color = STYLE.text_color
+
+STYLE = load_default_style()
+
+
+def get_style() -> StyleConfig:
+    """Return the current style configuration."""
+    return STYLE
+
+
+def set_style(style: StyleConfig):
+    """Set the current style configuration."""
```

Comment:
Let's also mention specifically that this is for setting the plot style globally? It wasn't immediately obvious to me (as a reader / user of the API)